lib/osf-components/addon/components/validated-model-form/component.ts

```diff
@@ -72,19 +80,34 @@ export default class ValidatedModelForm<M extends ValidatedModelName> extends Co
         super(...args);
 
         assert('Can only pass either a model or a modelName', !(Boolean(this.model) && Boolean(this.modelName)));
-
         if (!this.model && this.modelName) {
             this.model = this.store.createRecord(this.modelName, this.modelProperties);
         }
+        if (!this.changeset && this.model) {
+            set(this, 'changeset', this.buildChangeset(this.model));
+        }
     }
 
     willDestroy() {
-        if (this.model) {
-            if (this.onWillDestroy !== undefined) {
-                this.onWillDestroy(this.model);
-            } else if (!this.saved) {
-                this.model.unloadRecord();
-            }
+        if (this.changeset) {
+            this.changeset.rollback();
```

Comment:
Is rolling back necessary? I'd have assumed we could just let the changeset be garbage collected.

Reply:
Good point. I'll verify.

Reply:
Yup, this all works the way it should. Good catch!